File: src/main/java/io/sch/historyscan/domain/contexts/codebase/find/CodeBasesSearch.java
package io.sch.historyscan.domain.contexts.codebase.find;

import java.util.List;

public record CodeBasesSearch(CodeBasesListInventory codeBasesListInventory) implements FindCodeBases {

    @Override
    public List<CurrentCodeBase> fromDisk() {
        return codeBasesListInventory.listAll();
    }
}

File: historyscan-back/src/main/kotlin/io/sch/historyscan/domain/contexts/codebase/find/CodeBasesSearch.kt
package io.sch.historyscan.domain.contexts.codebase.find

class CodeBasesSearch(private val codeBasesListInventory: CodeBasesListInventory) : FindCodeBases {
    override fun fromDisk(): List<CurrentCodeBase> {
        return codeBasesListInventory.listAll()
    }
}
